src/Skia/Avalonia.Skia/Helpers/ImageSavingHelper.cs

```diff
@@ -14,31 +14,50 @@ public static class ImageSavingHelper
         /// </summary>
         /// <param name="image">Image to save</param>
         /// <param name="fileName">Target file.</param>
-        public static void SaveImage(SKImage image, string fileName)
+        /// <param name="quality">
+        /// The optional quality for PNG compression. 
+        /// The quality value is interpreted from 0 - 100. If quality is null 
+        /// the encoder applies the default quality value.
+        /// </param>
+        public static void SaveImage(SKImage image, string fileName, int? quality = null)
         {
             if (image == null) throw new ArgumentNullException(nameof(image));
             if (fileName == null) throw new ArgumentNullException(nameof(fileName));
 
             using (var stream = File.Create(fileName))
             {
-                SaveImage(image, stream);
+                SaveImage(image, stream, quality);
             }
         }
 
         /// <summary>
         /// Save Skia image to a stream.
         /// </summary>
         /// <param name="image">Image to save</param>
-        /// <param name="stream">Target stream.</param>
-        public static void SaveImage(SKImage image, Stream stream)
+        /// <param name="quality">
+        /// The optional quality for PNG compression. 
+        /// The quality value is interpreted from 0 - 100. If quality is null 
+        /// the encoder applies the default quality value.
+        /// </param>
+        public static void SaveImage(SKImage image, Stream stream, int? quality = null)
         {
             if (image == null) throw new ArgumentNullException(nameof(image));
             if (stream == null) throw new ArgumentNullException(nameof(stream));
 
-            using (var data = image.Encode())
+            if (quality == null)
             {
-                data.SaveTo(stream);
+                using (var data = image.Encode())
+                {
+                    data.SaveTo(stream);
+                }
+            }
+            else
+            {
+                using (var data = image.Encode(SKEncodedImageFormat.Png, (int)quality))
```

Comment:
Since PNG is a lossless format, I don't think quality parameter does anything here.

Reply:
It does though, it determines the size of the resulting file and it determines the time consumed for the encoding. More compression => smaller file size and longer time.

I am using this often (also for Gdi.Bitmap) for serialization (also base64) and there it seemed relevant to tweak it to specific needs.

I agree, this may not be so relevant for the sake of AvaloniaUI. However, using Avalonia.Bitmap within the UI may require someone (like me) to serialize and de-serialize. In my case a do that from a REST interface with json serialization.

I just thought "completing" the Save params just like "most" Bitmaps do would not hurt too much...

Reply:
If there is a different way to save an Avalonia.Bitmap with quality control (without creating an extra copy) that would also be good enough, I guess. It just seemed to me that all internals of what is behind the Avalonia.Bitmap are not exposed... (for good reasons though)

Reply:
I you find that too much breaking the current interfaces, that's fine too. It is not so essential.

Reply:
Use case explanation:
The specific app I have been working on, receives uncompressed video frames with 30 fps and displays that using WriteabelBitmap. The images come over a standard REST call. Best performance is reached of you set quality to 100 instead of its default value (whatever that is for the backend). And also it is best to de-serialize it directly into the Avalonia.Bitmap.

Jepp, very specific.

Reply:
@kekekeks : should we drop this PR?

Reply:
@llfab please don't drop it, it sounds useful to me - sorry we haven't got round to reviewing it properly, we're rather busy!